test/boost/aws_errors_test.cc

```diff
@@ -0,0 +1,445 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#define BOOST_TEST_MODULE object_storage
+
+#include <boost/test/unit_test.hpp>
+#include <seastar/core/sstring.hh>
+#include <utils/s3/aws_errors.hh>
+
+uint8_t singular = 1;
+uint8_t plural = 2;
+
+namespace boost::test_tools::tt_detail {
+template <>
+struct print_log_value<aws::aws_error_type> {
+    void operator()(std::ostream& os, const aws::aws_error_type& error_type) { os << static_cast<uint8_t>(error_type); }
+};
+} // namespace boost::test_tools::tt_detail
+
+static seastar::sstring build_xml_response(const std::string& exception, const std::string& message, const std::string& requestId, uint8_t style = singular) {
+    std::stringstream ss;
+
+    ss << R"(<?xml version="1.0" encoding="UTF-8"?>)";
+    if (style & plural) {
+        ss << "<OtherRoot> <Errors>";
+    }
+
+    ss << "<Error>"
+       << "<Code>" << exception << "</Code>"
+       << "<Message>" << message << "</Message>"
+       << "<Resource>/mybucket/myfile.bin</Resource>";
+
+    if (style & singular) {
+        ss << "<RequestId>" << requestId << "</RequestId>";
+    }
+
+    ss << "</Error>";
+
+    if (style & plural) {
+        ss << "</Errors>"
+           << "<RequestId>" << requestId << "</RequestId>"
+           << "</OtherRoot>";
+    }
+    return ss.str();
+}
+
+BOOST_AUTO_TEST_CASE(TestXmlErrorPayload) {
+    std::string message = "Test Message";
+    std::string requestId = "Request Id";
+    aws::aws_error error = aws::aws_error::parse(build_xml_response("IncompleteSignatureException", message, requestId));
```

Comment:
i'd drive these test cases with an array or even using `aws_error_map` and a loop. more compacted this way, and it's more obvious on the connection between the input and expected outputs) this way.

but i don't feel strong either way.

Reply:
done